### PR TITLE
Optional use of DHCP NTP settings

### PR DESCRIPTION
**Pull request by markterrill**
The manufacturer may well not want a random customer's DHCP NTP service to be trusted. time.google is reliable, whatever is in their $50 local router and upstream ISP is not as reliable.

### CODE REVIEW

File: mos.yml [hunk 1]
```diff
@@ -12,7 +12,8 @@ filesystem:
 config_schema:
   - ["sntp", "o", {title: "SNTP settings"}]
   - ["sntp.enable", "b", true, {title: "Enable SNTP"}]
-  - ["sntp.server", "s", "time.google.com", {title: "Server address. If DHCP lease contains NTP server, tries that first."}]
+  - ["sntp.server", "s", "time.google.com", {title: "Server address"}]
+  - ["sntp.follow_dhcp_ntp", "b", false, {title: "If DHCP lease contains NTP server, tries that first."}]
```
Comment (rojer) [inline mos.yml]: `sntp.use_dhcp_ntp_server`
and please make `true` the default to preserve current behavior.

File: src/mgos_sntp.c [hunk 2]
```diff
@@ -108,7 +108,7 @@ static bool mgos_sntp_query(const char *server) {
 static void mgos_sntp_retry_timer_cb(void *user_data) {
   s_state.retry_timer_id = MGOS_INVALID_TIMER_ID;
   const char *server = mgos_sys_config_get_sntp_server();
-  if (s_state.idx == 0 && s_state.local_server[0] != '\0') {
+  if (s_state.idx == 0 && s_state.local_server[0] != '\0' && mgos_sys_config_get_sntp_follow_dhcp_ntp()) {
```
Comment (rojer) [inline src/mgos_sntp.c]: please run `clang-format` on changed files.

**Review by rojer — CHANGES_REQUESTED** [review]

File: src/mgos_sntp.c (hunk 2 above)
Comment (markterrill) [reply src/mgos_sntp.c]: mate, that line is actually correct even with clang-format run on it. I've even added clang-format to the folder with another commit.

File: mos.yml (hunk 1 above)
Comment (markterrill) [reply mos.yml]: Done, though the whole point of this PR is to undo the undesirable change to behaviour introduced previously

File: src/mgos_sntp.c (hunk 2 above)
Comment (rojer) [reply src/mgos_sntp.c]: thanks for adding .clang-format.
it can't be ok though, google style specifies 80 column limit and this line is 100+

Comment (markterrill) [reply src/mgos_sntp.c]: CLion is saying it's fine.

File: mos.yml (hunk 1 above)
Comment (rojer) [reply mos.yml]: i think using local NTP is a sensible default for most use cases.
feel free to override the default in your project though.

File: src/mgos_sntp.c (hunk 2 above)
Comment (markterrill) [reply src/mgos_sntp.c]: wait a second, updating the lines manually and the config fetch to match new name

Comment (rojer) [reply src/mgos_sntp.c]: it's not though, either it's not using the `.clang-format` or something else is amiss.
ok, just break the line at `&&`, that's all that's needed anyway.

**Comment by markterrill** [comment]
@rojer have manually done new lines on those >80 col lines and realised the config_get needed updating to mgos_sys_config_get_sntp_use_dhcp_ntp_server

**Comment by rojer** [comment]
merged, thanks!

**Comment by rojer** [comment]
i fixed formatting in https://github.com/mongoose-os-libs/sntp/commit/fdea2f35a3d2fb20bc590960a053eefcd67600ba - this is what proper clang-formatting with this config looks like. i don't think your IDE does it, you may want to have a look.

**Pull request merged**